spec/models/bus_stop_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'spec_helper'
+
+describe BusStop do
+  let!(:pending_stop) { create :bus_stop, :pending }
+  let!(:completed_stop) { create :bus_stop, :completed }
+  let!(:not_started_stop) { create :bus_stop }
+  describe 'decide_if_completed_by' do
+    let(:user) { create :user }
+    let(:stop) { create :bus_stop, :pending }
+    context 'bus stop completed attribute changed' do
+      context 'bus stop is completed' do
+        it 'assigns user to completed by' do
+          stop.update! accessible: true
+          stop.completed = true
+          stop.decide_if_completed_by user
+          expect(stop.completed_by).to eql user
+        end
+      end
+      context 'bus stop is not completed' do
```

Comment:
This looks like a potential false positive. I do see that the stop is pending, and so not completed, but I *also* see that the completed hasn't changed.

Since we haven't fulfilled *either* condition under which the method runs, logic dictates that this isn't an exhaustive test.

Reply:
:( ok

Reply:
I've updated it?

Reply:
Yes, looks correct now.